metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableBackend.java

```diff
@@ -413,46 +421,78 @@ public Statistics getStatistics() {
     }
 
     private AsyncFuture<FetchData.Result> fetchBatch(
-        final FetchQuotaWatcher watcher, final MetricType type, final List<PreparedQuery> prepared,
-        final BigtableConnection c, final Consumer<MetricReadResult> metricsConsumer
+        final FetchQuotaWatcher watcher,
+        final MetricType type,
+        final List<PreparedQuery> prepared,
+        final BigtableConnection c,
+        final Consumer<MetricReadResult> metricsConsumer,
+        final Span parentSpan
     ) {
         final BigtableDataClient client = c.dataClient();
 
+        final Span fetchBatchSpan =
+            tracer.spanBuilderWithExplicitParent("bigtable.fetchBatch", parentSpan).startSpan();
+        fetchBatchSpan.putAttribute("preparedQuerySize", longAttributeValue(prepared.size()));
+
         final List<AsyncFuture<FetchData.Result>> fetches = new ArrayList<>(prepared.size());
 
         for (final PreparedQuery p : prepared) {
-            QueryTrace.NamedWatch fs = QueryTrace.watch(FETCH_SEGMENT);
-
-            final AsyncFuture<List<FlatRow>> readRows = client.readRows(table, ReadRowsRequest
-                .builder()
-                .range(new RowRange(Optional.of(p.rowKeyStart), Optional.of(p.rowKeyEnd)))
-                .filter(RowFilter.chain(Arrays.asList(RowFilter
-                    .newColumnRangeBuilder(p.columnFamily)
-                    .startQualifierOpen(p.startQualifierOpen)
-                    .endQualifierClosed(p.endQualifierClosed)
-                    .build(), RowFilter.onlyLatestCell())))
-                .build());
+            Span readRowsSpan = tracer.spanBuilderWithExplicitParent(
```

Comment:
Same comment as above for closing out this span.

Reply:
fixed 👍